### PR TITLE
gh-99767: update PyTypeObject docs for type watchers

### PR DESCRIPTION
**Pull request by carljm**
* Issue: gh-99767

### CODE REVIEW

File: Doc/c-api/typeobj.rst [hunk 1]
```diff
@@ -2090,6 +2092,15 @@ and :c:type:`PyType_Type` effectively act as defaults.)
    .. versionadded:: 3.9 (the field exists since 3.8 but it's only used since 3.9)
 
 
+.. c:member:: char PyTypeObject.tp_watched
```
Comment (markshannon) [inline Doc/c-api/typeobj.rst:2095]: I would rather the docs say "Internal. Do not use."

TBH, I would like the whole of the type struct to be "do not use", but let's not make things any worse.

Comment (carljm) [reply Doc/c-api/typeobj.rst:2095]: Are you wanting the existing text removed and replaced so it says _only_ "Internal, do not use"? Or do you want to add "Internal, do not use" but also keep the description of what it is?

Comment (carljm) [reply Doc/c-api/typeobj.rst:2095]: Pushed a version that removes the existing text, but I can easily re-add it.

**Comment by markshannon** [comment]
That looks good. Better to document it as "do not use", than have folks try to figure out how to (mis)use it.

File: Doc/c-api/typeobj.rst [hunk 2]
```diff
@@ -147,6 +147,8 @@ Quick Reference
    +------------------------------------------------+-----------------------------------+-------------------+---+---+---+---+
    | :c:member:`~PyTypeObject.tp_vectorcall`        | :c:type:`vectorcallfunc`          |                   |   |   |   |   |
    +------------------------------------------------+-----------------------------------+-------------------+---+---+---+---+
+   | :c:member:`~PyTypeObject.tp_watched`           | char                              |                   |   |   |   |   |
+   +------------------------------------------------+-----------------------------------+-------------------+---+---+---+---+
```
Comment (mattip) [inline Doc/c-api/typeobj.rst:151]: If this slot is internal, it should be in square brackets like `tp_version_tag`

Comment (carljm) [reply Doc/c-api/typeobj.rst:151]: Thanks, sorry I missed this. Added in #100271

Comment (mattip) [reply Doc/c-api/typeobj.rst:151]: Thanks

**Pull request merged**